format-common/src/main/java/io/cdap/plugin/format/plugin/AbstractFileSource.java

```diff
@@ -162,15 +165,28 @@ public void prepareRun(BatchSourceContext context) throws Exception {
     Pattern pattern = config.getFilePattern();
     if (schema == null) {
       SchemaDetector schemaDetector = new SchemaDetector(validatingInputFormat);
-      schema = schemaDetector.detectSchema(config.getPath(context), pattern,
-                                           formatContext, getFileSystemProperties(null));
+      try {
+        schema = schemaDetector.detectSchema(config.getPath(context), pattern,
+                                             formatContext, getFileSystemProperties(null));
+      } catch (IOException e) {
+        collector.addFailure("Error when trying to detect schema: " + e.getMessage(), null)
+          .withStacktrace(e.getStackTrace());
+        collector.getOrThrowException();
+      }
     }
     formatContext = new FormatContext(collector, schema);
     validateInputFormatProvider(formatContext, fileFormat, validatingInputFormat);
     validatePathField(collector, schema);
     collector.getOrThrowException();
 
-    Job job = JobUtils.createInstance();
+    Job job = null;
+    try {
+      job = JobUtils.createInstance();
+    } catch (IOException e) {
+      collector.addFailure("Failed to create job instance.", e.getMessage())
```

Comment:
`e.getMessage()` is not a corrective action:

```
collector.addFailure(String.format("Failed to create job instance, %s: %s", e.getClass().getName(), e.getMessage()), null)
```

This comment applies to whole PR.